client-tez/src/main/java/org/apache/tez/common/ShuffleAssignmentsInfoWritable.java

```diff
@@ -0,0 +1,160 @@
+
+
+public class ShuffleAssignmentsInfoWritable implements Writable {
+  private ShuffleAssignmentsInfo shuffleAssignmentsInfo;
+
+  public ShuffleAssignmentsInfoWritable() {
+
+  }
+
+  public ShuffleAssignmentsInfoWritable(ShuffleAssignmentsInfo shuffleAssignmentsInfo) {
+    this.shuffleAssignmentsInfo = shuffleAssignmentsInfo;
+  }
+
+
+  @Override
+  public void write(DataOutput dataOutput) throws IOException {
+    if (shuffleAssignmentsInfo == null) {
+      dataOutput.writeInt(-1);
+      return;
+    } else {
+      dataOutput.writeInt(1);
+    }
+
+    Map<Integer, List<ShuffleServerInfo>> partitionToServers = shuffleAssignmentsInfo.getPartitionToServers();
+    if (MapUtils.isEmpty(partitionToServers)) {
+      dataOutput.writeInt(-1);
+    } else {
+      dataOutput.writeInt(partitionToServers.size());
+      for (Map.Entry<Integer, List<ShuffleServerInfo>> entry : partitionToServers.entrySet()) {
+        dataOutput.writeInt(entry.getKey());
+        if (CollectionUtils.isEmpty(entry.getValue())) {
+          dataOutput.writeInt(-1);
+        } else {
+          dataOutput.writeInt(entry.getValue().size());
+          for (ShuffleServerInfo serverInfo : entry.getValue()) {
+            dataOutput.writeUTF(serverInfo.getId());
+            dataOutput.writeUTF(serverInfo.getHost());
+            dataOutput.writeInt(serverInfo.getGrpcPort());
+          }
+        }
+      }
+    }
+
+    Map<ShuffleServerInfo, List<PartitionRange>> serverToPartitionRanges = shuffleAssignmentsInfo
+            .getServerToPartitionRanges();
+    if (MapUtils.isEmpty(serverToPartitionRanges)) {
+      dataOutput.writeInt(-1);
+    } else {
+      dataOutput.writeInt(serverToPartitionRanges.size());
+      for (Map.Entry<ShuffleServerInfo, List<PartitionRange>> entry : serverToPartitionRanges.entrySet()) {
+        dataOutput.writeUTF(entry.getKey().getId());
+        dataOutput.writeUTF(entry.getKey().getHost());
+        dataOutput.writeInt(entry.getKey().getGrpcPort());
+        if (CollectionUtils.isEmpty(entry.getValue())) {
+          dataOutput.writeInt(-1);
+        } else {
+          dataOutput.writeInt(entry.getValue().size());
+          for (PartitionRange range : entry.getValue()) {
+            dataOutput.writeInt(range.getStart());
+            dataOutput.writeInt(range.getEnd());
+          }
+        }
+      }
+    }
+  }
+
+  @Override
+  public void readFields(DataInput dataInput) throws IOException {
+    if (dataInput.readInt() == -1) {
+      return;
+    }
+
+    Map<Integer, List<ShuffleServerInfo>> partitionToServers = new HashMap<>();
+    int partitionToServersSize = dataInput.readInt();
+    if (partitionToServersSize != -1) {
+      Integer k;
+      for (int i = 0; i < partitionToServersSize; i++) {
+        k = dataInput.readInt();
+        List<ShuffleServerInfo> v = new ArrayList<>();
+        int vSize = dataInput.readInt();
+        if (vSize != -1) {
+          for (int i1 = 0; i1 < vSize; i1++) {
```

Comment:
Could we have better name like `shuffleServerIndex`?

Reply:
ok, we modified it